packages/css/src/components/radio/radio.scss

```diff
@@ -44,111 +35,158 @@
   gap: var(--ams-radio-gap);
   line-height: var(--ams-radio-line-height);
   outline-offset: var(--ams-radio-outline-offset);
+  position: relative;
 
-  &:hover {
-    color: var(--ams-radio-hover-color);
-    text-decoration-line: underline;
-    text-decoration-thickness: var(--ams-radio-hover-text-decoration-thickness);
-    text-underline-offset: 0.375rem;
+  // Radio button check mark
+  &::after {
+    background-color: var(--ams-radio-button-checked-background-color);
+    block-size: calc(var(--ams-radio-font-size) * var(--ams-radio-line-height));
+    flex-shrink: 0;
+    inline-size: 1.5rem;
+    -webkit-mask-image: var(--ams-radio-button-checked-mask-image);
+    mask-image: var(--ams-radio-button-checked-mask-image);
+    -webkit-mask-position: center;
+    mask-position: center;
+    -webkit-mask-repeat: no-repeat;
+    mask-repeat: no-repeat;
+    position: absolute;
+
+    @include forced-colors;
+  }
 
-    .ams-radio__circle::after {
-      border-color: var(--ams-radio-circle-hover-border-color);
-    }
+  // Radio button
+  &::before {
+    background-color: var(--ams-radio-button-background-color);
+    block-size: calc(var(--ams-radio-font-size) * var(--ams-radio-line-height));
+    content: "";
+    flex-shrink: 0;
+    inline-size: 1.5rem;
+    -webkit-mask-image: var(--ams-radio-button-mask-image);
+    mask-image: var(--ams-radio-button-mask-image);
+    -webkit-mask-position: center;
+    mask-position: center;
+    -webkit-mask-repeat: no-repeat;
+    mask-repeat: no-repeat;
+
+    @include forced-colors;
   }
 
   @include text-rendering;
 }
 
-// Default checked
-.ams-radio__input:checked {
-  + .ams-radio__label .ams-radio__circle::after {
-    background-image: var(--ams-radio-circle-checked-background-image);
+// Default hover
+.ams-radio__label:hover {
+  color: var(--ams-radio-hover-color);
+  text-decoration-line: underline;
+  text-decoration-thickness: var(--ams-radio-hover-text-decoration-thickness);
+  text-underline-offset: 0.375rem;
```

Comment:
Not related to ams.link-appearance.text-underline-offset?

Reply:
I think it should be, I'll change it

Reply:
Sweet